packages/cli/src/commands/build.ts

```diff
@@ -337,10 +339,16 @@ async function doBuild(
     await setMonorepoDefaultSettings(cwd, workPath, projectSettings, output);
   }
 
+  const ignoreFilter = await getIgnoreFilter(workPath, rootDirectory);
+  function isAllowed(filePath: string) {
+    return !ignoreFilter(filePath);
+  }
+
   // Get a list of source files
-  const files = (await getFiles(workPath, client)).map(f =>
-    normalizePath(relative(workPath, f))
-  );
+  const rawFiles = await getFiles(workPath, client);
+  const files = rawFiles
+    .map(f => normalizePath(relative(workPath, f)))
+    .filter(isAllowed);
```

Comment:
IMO just inline this

```suggestion
    .filter(f => !ignoreFilter(f));
```